src/eko/output/struct.py

```diff
@@ -749,19 +749,12 @@ def detached(cls, theory: dict, operator: dict, path: pathlib.Path):
 
         """
         bases = operator["rotations"]
-        for basis in ("inputgrid", "targetgrid", "inputpids", "targetpids"):
-            if basis in bases:
-                bases[f"_{basis}"] = bases[basis]
-                del bases[basis]
         bases["pids"] = np.array(br.flavor_basis_pids)
-        for k in ("xgrid", "_inputgrid", "_targetgrid"):
-            if k in operator["rotations"]:
-                if operator["rotations"][k] is None:
-                    continue
-                bases[k] = interpolation.XGrid(
-                    operator["rotations"][k],
-                    log=operator["configs"]["interpolation_is_log"],
-                )
+        if operator["rotations"]["xgrid"] is not None:
```

Comment:
Actually, I'm not sure any longer with need this `if`: `xgrid` should not be allowed to be `None`.

Isn't it? @felixhekhorn